src/iOS/Avalonia.iOS/Storage/IOSStorageItem.cs

```diff
@@ -150,6 +150,7 @@ internal sealed class IOSStorageFolder : IOSStorageItem, IStorageBookmarkFolder
 {
     public IOSStorageFolder(NSUrl url) : base(url)
     {
+        Url.StartAccessingSecurityScopedResource();
```

Comment:
I don't think StartAccessingSecurityScopedResource should be global here for a whole lifetime of the folder. Not to mention, Dispose does nothing in this case.

Wouldn't it be better to add StartAccessingSecurityScopedResource + Stop to the specific operations? Like, GetItemsAsync.

Reply:
Yes, you are right. I was debating this with myself and made the wrong decision.

I started converting to per-operation calls, but I'm not sure how to handle StartAccessingSecurityScopedResource's return value. Should I return early when it fails like IOSStorageItem.SaveBookmarkAsync does, or should I just let it fail later? An argument for the latter is that a permission denied error would be propagated to the caller\* instead of just receiving null.

\* At least for GetItemsAsync and CreateFolderAsync. CreateFileAsync does not throw, because NSFileManager.DefaultManager.CreateFile does not give error information.

Reply:
I don't see any problem with not throwing early here, as iOS native exception might have more detailed information about the error. Or, alternatively, it can be wrapped as an inner exception.

But either way, if StartAccessingSecurityScopedResource returned false, we shouldn't call StopAccessingSecurityScopedResource then.

Reply:
Thanks, I went with iOS native exceptions. 

[Apple's documentation](https://developer.apple.com/documentation/foundation/nsurl/1413736-stopaccessingsecurityscopedresou) says "_If you call this method on a URL whose referenced resource you don’t have access to, nothing happens_.", so I think it's OK to call StopAccessingSecurityScopedResource even if the Start failed.

I hit a snag with MoveAsync, which led me to other problems with accessing content in opened folders. These all boil down to the fact that only the IOSStorageFolder directly opened with StorageProvider.OpenFolderPickerAsync has a security-scoped NSUrl, the IOSStorageItems acquired via CreateFileAsync, CreateFolderAsync and GetItemsAsync from this folder do not, even though we have access to them (UIDocumentPickerViewController gives recursive access) \*. In practice, this means calling StartAccessingSecurityScopedResource on such objects' Url fails; we need to call it on the security-scoped ancestor's Url. My solution stores this ancestor's Url in every IOSStorageItem, and calls Start/StopAccessingSecurityScopedResource on this instead of it's own.

\* _I'd like to clarify that this is not an Avalonia, or even Xamarin.iOS thing, this behaviour comes from Apple's APIs._